### ISSUE TITLE
Custom Fields in pro / post-widget

### ISSUE THREAD

**Issue by dasigna** [pojome/elementor#1071]
any chance or known usable way to get a grip on cpt-fields for post-widgets in pro???
tried with no luck... but would be some sort of 'highly recommended' much more than 'very nice to have' if one could choose those fields in 'meta-data' for 'posts'-widget, and even more for 'portfolio'!

btw. having more styling options for the latter would be even great. :-)

**Comment by joshmarom** [comment]
Could you please elaborate on how you think this should be incorporated into the widget?

**Comment by dasigna** [comment]
... what do you mean by 'elaborate'? i'm no coder, so 'elaborating' in its original meaning exceeds my abilities.

but in general i think there should be a way to show those fields in a way it is possible for post-titles or normal metas... (as one can do like choosing queries) if i had found a way to get a grip on that, i would have already done :-).

**Comment by joshmarom** [comment]
I'm not asking from the development point of view, let us worry about that. I mean, how do you imagine the UI of this feature and how do you imagine the output result to look like. What's the way you expect to use it?

I might agree with you that giving the option to use custom fields in portfolio and posts widgets is a good idea, but I don't know how exactly it can be introduced into the widget in terms of UI and in terms of output result.

Since you asked for it, I assume you have an idea on how you think it should be.

**Comment by dasigna** [comment]
hi josh,
glad i dont have to dig into development :-).

but lets see what i have in mind:
for portfolio it would just be great to have them (custom post type fields) in the same way as it is possible for title - just expanding overlay possibilities to show exerpt, custom fields etc. or just replacing title by some other field. (a static overlay apart from the hover state, with possibility to define the amount of covering -only half, or only text background etc would be even great btw)

for posts i am simply thinking of expanding the possibilities of selectable fields to show. as we can select title, meta and exerpt, this should also be possible for custom fields.
and: querys should also be possible for custom fields. custom taxonomies already appear, but there simply should be a way to query ctp fields. (btw i am missing color overlay for posts images here too :-) )

so in my opinion the existing ui might stay the same more or less - is just a matter of expanding the choices by getting a grip on cpt-fields (if there are).
hope that was a bit helpful for you - if not, please ask again.

so a little remark concerning this one:
as i am often using '[wp types](https://wp-types.com)' for creating ctps and there are already implementations in this plugin for other builders, there maybe is a way to contact them reagarding doing the same for elementor? (that would really be great, because creating frontend-views for cpts within types is a mess, and their own builder not worth mentioning...)

**Comment by sigersmit** [comment]
+1. I am using wp-types al lot also, would be great if in some way it would be possible to get custom fields to be shown with Elementor (pro)

**Comment by HelmutGoebel** [comment]
I agree with dasigna and sigersmit, although I don't use particularly wp-types, but heavily custom fields and taxonomies in custom post types. I want to give the websurfer standard informations (in my book-blog: number of pages, author, size, price e.a.), so he know in advance, if the post is interesting for her/him or not.

**Comment by joffcrabtree** [comment]
I would love to see this implemented for ANY custom field. It will completely blow other page builders away. This is my PRIMARY bug bear with Elementor and would love to see some attention focused on this fairly important issue

**Comment by dkenzik** [comment]
Agree with everyone regarding custom fields, notably @joffcrabtree. It would be a nice addition to have access to native cf, ACF and wp-types' custom fields (assuming it's not using native under the hood.)

**Comment by dasigna** [comment]
just have to pull this up again.

just facing another case where it just would be more than helpful to have custom fields for cpt's styled within elementor. even far more than a nice addition ... 

wp types custom fields e.g. are stored in native _postmeta table, so i think this should be quite comfortably possible to implement.
tried various ways to get them hooked via elementor with no success so far - sadly. maybe possible via custom shortcode or custom made widget, but that exeeds my personal abilities up til now.

ctps show up nicely - their fields should do the same :-)

hopefully elementor guys are still listening and find the chance to work on this soon (the sooner, the better). thanks.

**Comment by pulcix** [comment]
I'm here for support this cause too! :)

**Comment by joshmarom** [comment]
We are working on a new feature that will enable the use of CPT and other dynamic texts in certain text fields of different widgets. It is still work in progress, so please be patient.
Later on, we have some plans for a new flexible way to create a customizable loop, where you can fully customize the elements of the loop items, including adding CPT based content. This feature, however, is just at the concept stage and will take some time to develop.

In short, we are working on it, but it's going to take a while.

**Comment by dasigna** [comment]
i'll try to be patient ... :-)

**Comment by smatzis** [comment]
Thank you for working on this guys. I think that integration with pods will solve all of the above problems. Pods already does excellent custom post types and fields. I asked the same question in Pods Forum (elementor to integrate with pods) and this is what i got: 

> You can definitely reach out to those other Page Builders and see if they’d like to partner with us. The only reason we were able to do the Beaver Themer integration is because one of our Community Developers wanted to work on it. We don’t really have the resources to devote direct programming resources, but if someone is willing to take up the project, we’re happy to help them succeed.

_Post link:_ https://wordpress.org/support/topic/thank-you-1408/#post-9396551

**So is it possible to cooperate with them to bring elementor to a whole new level?**

**Comment by dasigna** [comment]
... those custon post types and fields should be generally able to call - no matter if they're from pods, acf, wptypes or created by our own!
they're all within the database... el just has to get a grip on them and us being able to style the output.

easy huh? :-)

**Comment by Ver3** [comment]
Following. Can't wait for this to become real. Would be very awesome indeed, If we could use an Elementor Front end form to fill a custom post type (like a Galley or even Woo products).

**Comment by dasigna** [comment]
being patient is beginning to hurt .)

**Comment by joffcrabtree** [comment]
Although it's not what some want to hear. Just buy Anywhere Elementor Pro. I bought the unlimited sites license during the Black Friday sales for practically chicken feed

**Comment by dasigna** [comment]
@joshmarom 
wasnt meant that sad, but lucky to hear its coming soon (hopefully with no delay) and i will praise the day its here :-)

@joffcrabtree 
indeed, didnt want to hear that :-).
the goal is to have it together with elementor and all the resulting benefits. not targeting towards more and more addons ... not to mention AEP doesnt cover all cenarios and is quite pricey - imho.

**Comment by carlosnpz** [comment]
I'd love a native funcionality or integration with Advanced Custom Fields plugin. Hope you guys from dev team find a good solution for that. Btw, you're already awesome! Elementor rocks!!!

**Comment by black-eye** [comment]
+1

**Comment by quinnfive** [comment]
+1

**Comment by maxwellhibbert** [comment]
Fields connections would be awesome. ACF and or Pods.

**Comment by stickypages** [comment]
I am currently writing a theme, that integrates as much as I can so far. 
It includes, integration with Elementor Default Colors, Default Fonts, Integration with WP-Types, and tons of speed enhancements... I'm developing a website with it now, to help work out any major kinks. Maybe it will help with those down the road if Elementor doesn't release.

**Comment by dasigna** [comment]
@stickypages 
... ??? ...

**Comment by stickypages** [comment]
@dasigna Just seeing if I can integrate the background-feature. It's what I was working on before I found this thread. Keep ya' posted.

**Comment by maxwellhibbert** [comment]
@stickypages Your theme sounds interesting - please share it when it's ready, but how does it connect Elementor widgets to custom fields?  I think that functionality will have to be baked in by the Elementor team..

**Comment by stickypages** [comment]
Yeah, i'm trying to extend the element, but it's posing some issues. I might have to find a work-around

**Comment by maxwellhibbert** [comment]
I'm hoping for functionality something like Beaver Themer where the content for elements can be linked to custom fields. There's no way I'd let clients into an Elementor page design to edit content, the design would be broken in a heartbeat, taking more time on my end than Elementor saved in the first place 😄   Is this the sort of thing we're talking about here?

**Comment by stickypages** [comment]
@maxwellhibbert I won't be able to provide that functionality, that would be more 'core'. Therefore I'm going to find a work-around to be able to apply it. Never a perfect scenario it seems. But it will work for me... until Elementor get's something in place.

**Comment by stickypages** [comment]
If anyone is interested, you can download the master/child theme here. I'm still working on integration. But perhaps it will help someone else out. I got frustrated with other themes, so I started making my own. In the short time, I wasn't able to get a functioning fix for the "Featured Image". Except for adding a CSS class to your 'section' called, "featured-image" and it will apply the featured image to that section using the same settings in the Elementor element.

http://www.stickypages.ca/external_dependants/Sticky-master.zip
http://www.stickypages.ca/external_dependants/Sticky-child.zip

**Comment by maxwellhibbert** [comment]
Would it be possible for someone from the Elementor team to let us know what to expect when it comes to field connections in the upcoming 2.0 release?

I'm about to build a large-ish site for a client and would love to use Elementor over the Beaver because it's better/faster/stronger is most ways, but I'll need field connections for text, images and galleries eventually (could wait until 2.0 release).

**Comment by dasigna** [comment]
i'd like to know this too for the same reason as @maxwellhibbert !

... and even more: how about the custom queries???! (for grabbing future posts / showing events-> ref. #1748)
high, high hopes on this.

thanks.

**Comment by black-eye** [comment]
same curiousity and high hopes as @dasigna and @maxwellhibbert (and probably everybody else here)

**Comment by gnowland** [comment]
Are the custom fields available in 2.0.3?

**Comment by maxwellhibbert** [comment]
@gnowland In the [Pro V2 Beta](https://github.com/pojome/elementor/issues/3826)

**Comment by arielk** [comment]
We added this feature to E-Pro 2.0.0

Thanks for your feedback!

**Comment by gnowland** [comment]
Hi @arielk! I was hoping my custom meta fields would be available for use in the "Post Info" widget, under "Meta Data", is this on the roadmap?

**Comment by HelmutGoebel** [comment]
The video widget in 2.0 version is great, but the cover image should be dynamic as well.

**Comment by notechup** [comment]
@gnowland That's what I am looking for too. A way to add Custom ACF fields to the Meta data of the post widget.

Case example: I have Custom Event Posts and I want to show when the event Starts as opposed to when the post was published, using the meta data. Not sure what the alternative is.

**Comment by scorpiknox** [comment]
Is this issue really closed? @arielk says this feature was included in E-Pro 2.0.0 but I am not seeing it.

**Comment by kristiangl** [comment]
> Case example: I have Custom Event Posts and I want to show when the event Starts as opposed to when the post was published, using the meta data. Not sure what the alternative is.

@notechup – haha exact same situation here, I want to list the custom field event date in the post widget. 

Thinking about it, it would probably be better to add ACF fields below to widget, instead of, or as well as in the meta section. I can see use cases where the meta should be separate to the ACF fields.

**Comment by joelpaul1** [comment]
It would make a great deal of difference if we could get Custom Fields as part of the Posts widget and not as separate widgets, still waiting for this to happen...

**Comment by alirexa** [comment]
I can't find this feature in pro version yet! why this issue is closed?!!

**Comment by alirexa** [comment]
is this closed?!

**Comment by danielporto** [comment]
Yes I agree, how is it closed if the problem was not solved?

**Comment by rudymacias** [comment]
Agree not seeing this either in E-Pro (using v2.1.9).

**Comment by kristiangl** [comment]
Just got word that this is being addressed in issue #4440, for anyone else who stumbles upon this issue.

**Comment by alirexa** [comment]
#4440 is not exactly the same, I think this is a separate feature! 
and the other one is also closed! :/

**Comment by kristiangl** [comment]
#4440 is still open. I can’t be sure, but I would say, logistically, it is too difficult to just add a custom field to the posts widget because of the wide range of field types. A custom loop template solves it for me, so I’m super happy, but it’s obviously a more complex solution.

**Comment by alirexa** [comment]
That would be a much better solution but I think we may see the feature in a much later update because of its' complexity.
and the weird thing is @arielk said : 
> We added this feature to E-Pro 2.0.0
> 
> Thanks for your feedback!
and nobody referenced this to the #4440.

**Comment by gnowland** [comment]
I received an email from Ben Pines today touting the addition of this specific feature to Elementor, and here's the blog post describing the release: https://elementor.com/custom-fields-integration/

This would be fantastic news -- can anyone separately confirm the update fulfills this issue's feature request? (I unfortunately can't confirm myself at this time)

**Comment by kristiangl** [comment]
I don’t think so. I haven’t had an update for Pro. I think they have just added custom field support to a bunch of basic widgets.

**Comment by vijaykr0** [comment]
I need to access a custom post meta field I added to my form in my post widget. The meta data drop down on Elementor pro contains only Author, Date, Time and Comments. I want to display the custom field under the excerpt in the widget and have a specific label added to it. How do I achieve that? 

Thanks,
V

**Comment by Otakumouse** [comment]
Super excited with the CUSTOM LOOP that was mentioned in this thread. Imagine being able to design a "single" post element that will be then repeated on the page depending on the query you set.

Any ETA on when is this going to be available?

**Comment by jpedigo** [comment]
Still don't have this feature and still not knowing why this is closed. This would definitely save me time having to code custom archive templates. Note that Oxygen does have this feature, though you have to know a little php (the get_post_meta call) to get it to work.

I was thinking it would be cool to use the existing custom post functionality to create a template that lays out exactly how you would want a post entry to look like, then be able to specify that template in the post grid widget as the way to render each post.

**Comment by jordanjustkidding** [comment]
I'm also looking to include a meta label showing the post type in my post grid layout. For example, posts that are in the custom post type 'event' have a label underneath the title (in the post grid) saying 'Event'. Has anyone figured a way of using custom hooks for this? Or has this been introduced into Elementor Pro yet? I'm not seeing it.

**Comment by LucasMatos88** [comment]
I also wish custom fields in portfólio and post widgets. 

An example will be an listing real state site that can display the city, the price and if the property is for sell, location or booth. 

Another example will be in portfólio see the client or the project type. 

That will expand infinety the types and flexibility of the website we can create with elementor.

**Comment by harunalikadic** [comment]
+1

**Comment by zahinwadud** [comment]
Still waiting for this feature. (Why this conversation is closed!)
Have you guys found any workaround / alternative approach for this?

**Comment by HelmutGoebel** [comment]
Yes. Use the Plugin "Ele Custom Skin". It gives you one more option in the themebuilder, called "Loop". There you can design a custom layout for a post grid.

![image](https://user-images.githubusercontent.com/14095280/61830640-ee0bdf80-ae6b-11e9-9e2b-d215d50ace0b.png)

In the posts widget, you'll find a 3rd option, called "custom". There you can select from the "Loop"-Templates available.

![image](https://user-images.githubusercontent.com/14095280/61830874-8f933100-ae6c-11e9-9b3a-d849060820f2.png)

Very easy and without any extra costs.

**Comment by jahanaydma** [comment]
Thanks @HelmutGoebel, but I'm experiencing issues with Ele Custom skin. I wish the custom fields were built in.

**Comment by HelmutGoebel** [comment]
So do I.

**Comment by revr0t** [comment]
> I also wish custom fields in portfólio and post widgets.
> 
> An example will be an listing real state site that can display the city, the price and if the property is for sell, location or booth.
> 
> Another example will be in portfólio see the client or the project type.
> 
> That will expand infinety the types and flexibility of the website we can create with elementor.

Exactly what I am looking for: > "An example will be an listing real state site that can display the city, the price and if the property is for sell, location or booth."
Any solutions out there?

**Comment by dasigna** [comment]
'Ele Custom Skin' is the way to go!
Use it on several installations so far - working great, no issues.

**Comment by davidamarek** [comment]
+1
